src/Client.php

```diff
@@ -117,6 +119,24 @@ public function setConfig(array $config): self
         return $this;
     }
 
+    /**
+     * @throws AuthenticationException
+     */
+    public function getElasticSearchClient(): \Elastic\Elasticsearch\Client
+    {
+        $clientBuilder = ClientBuilder::create();
```

Comment:
Is there any concern around recreating the builder each time or should we store in somewhere in the Client under `_elasticsearchClient?

Reply:
No, not really. If you look at the `create()` function, it just returns a new instance of the class, without any arguments or state. We could store it somewhere in a property as well. Would it make it sense to store in a property if there's no real state attached to the object?

Reply:
If there is no state, no need to store it. One thing I want to make sure, we don't open a new connection to Elasticsearch if we send multiple requests.

Reply:
I think (expect) all the connections the Elasticsearch Client makes will be opened and closed within the function call. The ClientBuilder - afaik - doesn't manage some sort of connection pool, if that's the sort of thing you were looking for?

Reply:
Found the part I was referring to here: https://github.com/ruflin/Elastica/blob/8.x/src/Transport/Http.php#L80 I don't think this is impacted by this change at all.